File: android/DFM/src/main/java/com/dontflymoney/watchers/ValueWatcher.java
package com.dontflymoney.watchers;

import com.dontflymoney.entities.Move;
import com.dontflymoney.viewhelper.AfterTextWatcher;

public class ValueWatcher extends AfterTextWatcher
{
	private Move move;

	public ValueWatcher(Move move)
	{
		this.move = move;
	}

	@Override
	public void textChanged(String text)
	{
		try { move.Value = Double.parseDouble(text); }
		catch (NumberFormatException ignored) { }
	}
}

File: android/DFM/src/main/kotlin/com/darakeon/uiHelpers/watchers/ValueWatcher.kt
package com.dontflymoney.watchers

import com.dontflymoney.entities.Move
import com.dontflymoney.viewhelper.AfterTextWatcher

class ValueWatcher(private val move: Move) : AfterTextWatcher() {

    override fun textChanged(text: String) {
        try {
            move.Value = java.lang.Double.parseDouble(text)
        } catch (ignored: NumberFormatException) {
        }

    }
}
